### DISCUSSION TITLE
HelmRelease not Ready if source-controller and helm-controller deployed in parallel

### DISCUSSION THREAD

**Discussion by hoffimar** [fluxcd/flux2#3152] [Q&A] [answered]
We have several helmchart and helmrelease resources existing in a k8s cluster, all in state Ready.
We're deploying both helm-controller and source-controller from the same helmchart built from the [manifests](https://github.com/fluxcd/flux2/releases/download/v0.34.0/manifests.tar.gz) with helm. When deploying a new (i.e. a different) version of this flux2 helm chart (containing helm-controller and source-controller), both controller pods are created in parallel. 

Now the problem description: after starting the **source-controller**, it first marks all helmcharts as not ready (error `no artifact available for HelmRepository source`), only after fetching an index of the HelmRepository the HelmCharts get status ready=true (log message `stored fetched index of size ...`), which takes a few seconds.
Unfortunately the **helm-controller** gets started in parallel, and when getting notified about the helmcharts being not ready, flags the HelmReleases as not ready. Since we set the reconcilation interval to 1h, the helmreleases are not ready until again reconciled. 

How can this situation be avoided? I can imagine splitting up our flux2 helm chart into separate ones (1 for source-controller, 1 for helm-controller) and thereby deploying both controllers sequentially could help, or use helm hooks to deploy them sequentially.
Or would you usually set the HelmReleases `interval` to a much smaller value like 1min? Maybe there is also some option to set for the controllers?

**Comment by kingdonb** [comment, marked as answer]
Linking to the issue where we are tracking this, I've replied there (it's not an identical report but it's very close, I think it may be the same root issue):

https://github.com/fluxcd/source-controller/issues/924
